### ISSUE TITLE
False positive with opening_braces when using if/let/first on a collection

### ISSUE THREAD

**Issue by TomQDRS** [realm/SwiftLint#3756]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

We are using the rule "opening_braces", but we are receiving a false positive when using if/let/first (see sample at the bottom). Removing the "where" condition in "first" removes the warning, leading me to believe that the curly braces inside `.first(where: {...})` are being falsely interpreted by swiftlint.

##### Output when running SwiftLint

```bash
$ swiftlint lint
[...]
/Users/redacted/Documents/[...]/redacted.swift:19:9: warning: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
[...]
```

### Environment

* SwiftLint version?
0.45.0
* Installation method used?
Homebrew
* Paste your configuration file:

```yml
disabled_rules: # rule identifiers to exclude from running
    - function_body_length
    - leading_whitespace
    - trailing_whitespace
    - cyclomatic_complexity
    - line_length
    - nesting
opt_in_rules: # some rules are only opt-in
    - empty_count
    - force_unwrapping
    - redundant_nil_coalescing
    - vertical_whitespace_closing_braces
    # - multiline_arguments # too many warnings
    # - multiline_arguments_brackets # too many warnings
    - multiline_parameters
    - multiline_parameters_brackets
excluded: # paths to ignore during linting. overridden by `included`.
    - Carthage
    - Pods
    - [REDACTED, but doesn't exclude the file with the violation]
# parameterized rules can be customized from this configuration file
# line_length: 220
# parameterized rules are first parameterized as a warning level, then error level.
type_body_length:
    - 300 # warning
    - 400 # error
# parameterized rules are first parameterized as a warning level, then error level.
file_length:
    - 800 # warning
    - 1000 # error
# parameterized rules are first parameterized as a warning level, then error level.
identifier_name:
    max_length:
        warning: 40
        error: 60
    min_length:
        warning: 2
        error: 1
vertical_whitespace:
    max_empty_lines: 1
#cyclomatic_complexity:
#    warning: 8
#    error: 14

custom_rules:
    multiple_conditionals_on_single_line:
        included: ".*.swift"
        regex: '(?:if |guard )(?:[\w =!<>]+[,&|]{1,2}[\w =!<>]+){2,}'
        severity: warning
        message: "There should not be three or more consecutive conditionals on a single line, please separate them with a linebreak."
    multiple_conditionals_grouping:
        included: ".*.swift"
        regex: '(?:if |guard )(?:[\w =!<>.?():]+[,&|]{1,2} *[\n\t]){1}'
        severity: warning
        message: "Conditionals separated by linebreaks should not be on the same line with guard/if."
    IBOutlets_should_be_weak:
        included: ".*.swift"
        name: "IBOutlets weak notification"
        regex: '@IBOutlet\s+var\s+((?!weak).+:)'
        message: "Should be weak to avoid retain cycles."
        severity: warning
    Avoid_Nil-Coalescing_in_strings:
        included: ".*.swift"
        name: "Nil-Coalescing in strings"
        regex: '\\\([\"\w\d\.]+\s\?{2}\s[\"\w\d\.]+\)'
        message: "Nil-Coalescing increased the build time dramatically"
        severity: warning
    fatal_error:
        included: ".*.swift"
        name: "Don't use fatal error"
        regex: "\bfatalError\b"
        message: "Don't use fatal errors."
        severity: error
    vertical_whitespace_after_opening_braces:
        included: ".*.swift"
        regex: '(?<!:\s)(class|extension)(?=\s)(?!\sfunc|\svar)[^\n]*\{\n[^\n]*[^\s]'
        severity: warning
        message: "There should be a vertical whitespace between declaration of a class/extension and its first member."
    superfluous_return:
        included: ".*.swift"
        regex: '(var|let)\h*\w*:\h*\w*\h*\{\h*return.*\}'
        severity: warning
        message: "There should be no return when you have a single line computed property."
    single_return_on_line_with_else:
        included: ".*.swift"
        regex: 'guard[ a-zA-Z0-9=?.\(\),><!\n]*else\s*\{(\s*[\n\r]+?\s*)return(\s*[\n\r]*?\s*)*\}'
        severity: warning
        message: "When returning from a guard/else clause without a return value or previous statements, the single return should be on the same height as the else."
    mark_separators:
        included: ".*.swift"
        regex: '\/\/ MARK: \-[ a-zA-Z0-9=?.\(\),><!-]*-\h*\v'
        severity: warning
        message: "Only use the opening separator for Marks."

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No
* Which Xcode version are you using?
13.1, Build Version 13A1030d
* Do you have a sample that shows the issue?
```swift
        if
            let a = ["A", "B"].first(where: { $0 == "A" }), // removing the parameter condition of "first" here removes the warning
            let b = ["B"].first
        { // This triggers the violation
            print(a)
        }
```

**Comment by TomQDRS** [comment]
Issue persists with 0.45.1.
